### PR TITLE
[SRVOCF-515] Add Node.js functions docs on arbitrary data types to the

### PR DESCRIPTION
**Pull request by rh-max**
Version(s):
`serverless-docs-1.33`+

Issue:
https://issues.redhat.com/browse/SRVOCF-515

Link to docs preview:
https://77513--ocpdocs-pr.netlify.app/openshift-serverless/latest/functions/reference/serverless-developing-nodejs-functions.html#serverless-nodejs-functions-context-objects_serverless-developing-nodejs-functions

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

**Comment by jrangelramos** [comment]
QE Approved

File: modules/serverless-nodejs-functions-context-objects.adoc [hunk 1]
```diff
@@ -53,3 +53,64 @@ function handle(context, data)
 ----
 
 The `data` parameter in this example is a JavaScript object that contains the `customerId` and `productId` properties.
+
+[id="serverless-nodejs-functions-context-objects-arbitrary-data_{context}"]
+== Arbitrary data
+
+A function can receive any data, not just CloudEvents. For example, you might want to call a function using POST with an arbitrary object in the body:
```
Comment (maxwelldb) [inline modules/serverless-nodejs-functions-context-objects.adoc:60]: Should "CloudEvents" be styled like an API object?

File: modules/serverless-nodejs-functions-context-objects.adoc [hunk 2]
```diff
@@ -53,3 +53,64 @@ function handle(context, data)
 ----
 
 The `data` parameter in this example is a JavaScript object that contains the `customerId` and `productId` properties.
+
+[id="serverless-nodejs-functions-context-objects-arbitrary-data_{context}"]
```
Comment (maxwelldb) [inline modules/serverless-nodejs-functions-context-objects.adoc:57]: This is probably an opportunity to modularize.

File: modules/serverless-nodejs-functions-context-objects.adoc (hunk 1 above)
Comment (maxwelldb) [reply modules/serverless-nodejs-functions-context-objects.adoc:60]: "using" -> "by using"

File: modules/serverless-nodejs-functions-context-objects.adoc [hunk 3]
```diff
@@ -53,3 +53,64 @@ function handle(context, data)
...
+In this case, you can define the function as follows:
+
+[source,javascript]
+----
+function handle(context, customer) {
+  return "Hello " + customer.contact.title + " " + customer.contact.lastname;
+}
+----
+
+Supplying the contact object to the function would then return the following:
```
Comment (maxwelldb) [inline modules/serverless-nodejs-functions-context-objects.adoc:83]: "following" should be an adjective in this case--"following example," or maybe "following example output."

File: modules/serverless-nodejs-functions-context-objects.adoc [hunk 4]
```diff
@@ -53,3 +53,64 @@ function handle(context, data)
...
+
+[source,text]
+----
+Hello Mr. Smith
+----
+
+[id="serverless-nodejs-functions-context-objects-supported-data-types_{context}"]
+== Supported data types
```
Comment (maxwelldb) [inline modules/serverless-nodejs-functions-context-objects.adoc:91]: This looks like a reference module.

File: modules/serverless-nodejs-functions-context-objects.adoc [hunk 5]
```diff
@@ -53,3 +53,64 @@ function handle(context, data)
...
+
+* *JSON Data*: Provided as a JavaScript object.
+* *XML Data*: Provided as an XML document.
+* *Plain Text*: Provided as a string.
+* *Binary Data*: Provided as a Buffer object.
+
+[id="serverless-nodejs-functions-context-objects-handling-different-data-types_{context}"]
+== Handling different data types
+
+Ensure your function can handle different data types by checking the Content-Type header and parsing the data accordingly. For example:
```
Comment (maxwelldb) [inline modules/serverless-nodejs-functions-context-objects.adoc:103]: This line says "procedure" to me, but there's no specific action to take, right?

**Review by maxwelldb — COMMENTED** [review]
Left some comments and suggestions. Re-label for another review if desired.

File: modules/serverless-nodejs-functions-context-objects.adoc (hunk 2 above)
Comment (rh-max) [reply modules/serverless-nodejs-functions-context-objects.adoc:57]: I'd also lean towards splitting this, but this is currently an (unwritten) convention for this type of reference docs in Serverless. It's better to do a full sweep for all such modules separately.

File: modules/serverless-nodejs-functions-context-objects.adoc (hunk 4 above)
Comment (rh-max) [reply modules/serverless-nodejs-functions-context-objects.adoc:91]: Same as above, this is better deferred to a mass restructuring of all such modules.

**Pull request merged**